compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/nballerina/nBallerinaCaller.java

```diff
@@ -0,0 +1,107 @@
+package org.wso2.ballerinalang.compiler.nballerina;
+
+import io.ballerina.runtime.api.PredefinedTypes;
+import io.ballerina.runtime.internal.configurable.providers.toml.TomlDetails;
+import io.ballerina.runtime.internal.launch.LaunchUtils;
+import io.ballerina.runtime.internal.scheduling.Scheduler;
+import io.ballerina.runtime.internal.scheduling.Strand;
+import io.ballerina.runtime.internal.util.exceptions.BLangRuntimeException;
+import io.ballerina.runtime.internal.util.exceptions.BallerinaException;
+import io.ballerina.runtime.internal.values.ErrorValue;
+import io.ballerina.runtime.internal.values.FutureValue;
+import org.ballerinalang.compiler.CompilerOptionName;
+import org.wso2.ballerinalang.compiler.tree.BLangPackage;
+import org.wso2.ballerinalang.compiler.util.CompilerContext;
+import org.wso2.ballerinalang.compiler.util.CompilerOptions;
+
+import java.io.File;
+import java.io.PrintStream;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Path;
+import java.util.function.Function;
+
+/**
+ * Call nBallerina backend.
+ */
+public class nBallerinaCaller {
+    private  final PrintStream console = System.out;
+    private static final CompilerContext.Key<nBallerinaCaller> NBALLERINA = new CompilerContext.Key<>();
+    private String nBal;
+
+    public static nBallerinaCaller getInstance(CompilerContext context) {
+        nBallerinaCaller nbalCaller = context.get(NBALLERINA);
+        if (nbalCaller == null) {
+            nbalCaller = new nBallerinaCaller(context);
+        }
+        return nbalCaller;
+    }
+
+    private nBallerinaCaller(CompilerContext context) {
+        context.put(NBALLERINA, this);
+        CompilerOptions compilerOptions = CompilerOptions.getInstance(context);
+        this.nBal = compilerOptions.get(CompilerOptionName.NBALLERINA);
+    }
+
+    public  void callnBallerina(BLangPackage pkgNode)  {
+        if (nBal != null) {
+            File path = new File(nBal);
+            try {
+                URLClassLoader cl = new URLClassLoader(new URL[]{path.toURI().toURL()});
+
+                Class<?> c = cl.loadClass("nballerina");
+                Method m = c.getMethod("compile", Strand.class);
+
+                Function<Object[], Object> func = objects -> {
+                    try {
+                        return m.invoke(null, objects[0]);
+                    } catch (InvocationTargetException e) {
+                        Throwable targetException = e.getTargetException();
+                        if (targetException instanceof RuntimeException) {
+                            throw (RuntimeException) targetException;
+                        } else {
+                            throw new RuntimeException(targetException);
+                        }
```

Comment:
If we use `throw new BallerinaException("Error invoking nBallerina backend", e);` will it print correct cause ?
if so we can use that, if not we can still throw `BallerinaException` don't think `RuntimeException` is the correct thing to throw from here.